src/charm.py

```diff
@@ -539,43 +584,42 @@ def _run_s3_migration(self) -> None:
             logger.exception("S3 migration failed with code %d.", cmd_err.exit_code)
             raise
 
-    def _set_up_discourse(self, event: HookEvent) -> None:
+    def _set_up_discourse(self) -> None:
         """Run Discourse migrations and recompile assets.
 
         Args:
             event: Event triggering the handler.
         """
-        container = self.unit.get_container(SERVICE_NAME)
+        container = self.unit.get_container(CONTAINER_NAME)
         if not self._are_relations_ready() or not container.can_connect():
-            logger.info("Defer discourse setup.")
-            event.defer()
+            logger.info("Pebble or relations not ready, not attempting to setup discourse")
             return
         logger.info(
             "Relations are ready and can connect to container, attempting to set up discourse."
         )
         try:
-            logger.info("Discourse setup: about to execute migrations.")
+            logger.info("Discourse setup: about to execute migrations")
             self._execute_migrations()
-            logger.info("Discourse setup: about to compile assets.")
+            logger.info("Discourse setup: about to compile assets")
             self._compile_assets()
-            logger.info("Discourse setup: about to mark the discourse setup process as complete.")
+            logger.info("Discourse setup: about to mark the discourse setup process as complete")
             self._set_setup_completed()
-            logger.info("Discourse setup complete, now setting workload version.")
+            logger.info("Discourse setup: about to set workload version")
             self._set_workload_version()
-            logger.info("Workload version set.")
+            logger.info("Discourse setup: completed")
         except ExecError as cmd_err:
             logger.exception("Setting up discourse failed with code %d.", cmd_err.exit_code)
             raise
 
-    def _config_changed(self, _: HookEvent) -> None:
+    def _change_config(self) -> None:
```

Comment:
This seems quite a confusing function name to me - it reads as if it's changing configuration. But the docstring suggest something else. How about something like `def _configure_pod`?

Reply:
I changed it to _configure_pod